paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -482,18 +482,36 @@ public void createTag(String tagName, long fromSnapshotId) {
                 snapshot != null,
                 "Cannot create tag because given snapshot #%s doesn't exist.",
                 fromSnapshotId);
-        createTag(tagName, snapshot);
+        return snapshot;
+    }
+
+    @Override
+    public void createTag(String tagName, long fromSnapshotId) {
+        createTag(tagName, coreOptions().tagTimeRetained(), createTagInternal(fromSnapshotId));
+    }
+
+    @Override
+    public void createTag(String tagName, Duration timeRetained, long fromSnapshotId) {
+        createTag(tagName, timeRetained, createTagInternal(fromSnapshotId));
     }
 
     @Override
     public void createTag(String tagName) {
```

Comment:
We should be very careful, old way should have no other fields in the json.

Reply:
good, thanks very much!
I have change TAG_DEFAULT_TIME_RETAINED to default none. And In TagManager.createTag() when createTag is null then create old snapshot json.